docs_src/microservices/application/V2Migration.md

```diff
@@ -0,0 +1,296 @@
+    ```toml
+      [Clients]
+        # Used for version check on start-up
+        # Also used for DeviceSevice, DeviceProfile and Device clients
+        [Clients.core-metadata]
+        Protocol = 'http'
+        Host = 'localhost'
+        Port = 59881
+        
+        # Used for Event client which is used by PushToCoreData function
+        [Clients.core-data]
+        Protocol = 'http'
+        Host = 'localhost'
+        Port = 59880
+        
+        # Used for Command client
+        [Clients.core-command]
+        Protocol = 'http'
+        Host = 'localhost'
+        Port = 59882
+        
+        # Used for Notification and Subscription clients
+        [Clients.support-notifications]
+        Protocol = 'http'
+        Host = 'localhost'
+        Port = 59860
+    ```
+
+#### Trigger
+
+The `Trigger` section (previously named `Binding`) has been restructured with `EdgexMessageBus` (previously named `MessageBus`) and `ExternalMqtt` (previously named `MqttBroker` ) moved under it. The `SubscribeTopics` (previously named `SubscribeTopic`) has been moved under the `EdgexMessageBus.SubscribeHost` and `ExternalMqtt` sections. The `PublishTopic` has been moved under the `EdgexMessageBus.PublishHost` and `ExternalMqtt` sections.
+
+##### EdgeX MessageBus
+
+If your Application Service is using the EdgeX MessageBus trigger, you can simply copy the complete `Trigger` configuration from the example below and tweak it as needed. 
+
+!!! example "Example - EdgeX MessageBus trigger configuration"
+
+    ```toml
+    [Trigger]
+    Type="edgex-messagebus"
+      [Trigger.EdgexMessageBus]
+      Type = "redis"
+        [Trigger.EdgexMessageBus.SubscribeHost]
+        Host = "localhost"
+        Port = 6379
+        Protocol = "redis"
+        SubscribeTopics="edgex/events/#"
+        [Trigger.EdgexMessageBus.PublishHost]
+        Host = "localhost"
+        Port = 6379
+        Protocol = "redis"
+        PublishTopic="example"
+        [Trigger.EdgexMessageBus.Optional]
+        AuthMode = 'usernamepassword'  # required for redis messagebus (secure or insecure).
+        SecretName = 'redisdb'
+    ```
+
+From the above example you can see the improved structure and the following changes:
+
+- Default `EdgexMessageBus` type has changed to `Redis` from `ZeroMQ`
+- Type value for `Redis` has changed from `redistreams` to `redis`. This is because the implementation no longer uses Redis Streams. It now using Redis Pub/Sub.
+- `SubscribeTopics` is now plural since it now accepts a comma separated list of topics. The default value uses a multi-level topic with a wild card. This is because Core Data and Device Services now publish to a multi-level topics which have`edgex/events` as their base. This allows Application Services to filter by topic rather then receive the data and then filter it out via a pipeline filter function. See the [Filter By Topics](../Triggers/#filter-by-topics) section for more details.
```

Comment:
publish to a multi-level _topic_

Reply:
Plural is correct in this case. The multi-level topics differ based on core data vs device service and on the event metadata. This is all covered in the referenced `Filter By Topics` section.